package.json

```diff
@@ -186,6 +186,61 @@
       "type": "object",
       "title": "Azure IoT Edge configuration",
       "properties": {
+        "azure-iot-edge.version.edgeHub": {
+          "type": "string",
+          "default": "1.0",
+          "description": "Set the edgeHub image version, which will be referenced in deployment manifest."
+        },
+        "azure-iot-edge.version.edgeAgent": {
+          "type": "string",
+          "default": "1.0",
+          "description": "Set the edgeAgent image version, which will be referenced in deployment manifest."
+        },
+        "azure-iot-edge.version.tempSensor": {
+          "type": "string",
+          "default": "1.0",
+          "description": "Set the tempSensor image version, which will be referenced in deployment manifest."
+        },
+        "azure-iot-edge.version.cmodule": {
+          "type": "string",
+          "default": null,
+          "description": "Set the C module template version, which will be refernced during scaffolding C module. null for latest."
+        },
+        "azure-iot-edge.version.pythonmodule": {
+          "type": "string",
+          "default": null,
+          "description": "Set the Python module template version, which will be refernced during scaffolding Python module. null for latest."
+        },
+        "azure-iot-edge.version.csharpmodule": {
+          "type": "string",
+          "default": null,
+          "description": "Set the C# module template version, which will be refernced during scaffolding C# module. null for latest."
+        },
+        "azure-iot-edge.templateInstall.csharpmodule": {
+          "type": "boolean",
+          "default": true,
+          "description": "Switch to install C# module template before scaffolding. Default is true."
+        },
+        "azure-iot-edge.templateInstall.csfunctionmodule": {
+          "type": "boolean",
+          "default": true,
+          "description": "Switch to install C# function module template before scaffolding. Default is true."
+        },
+        "azure-iot-edge.version.csfunctionmodule": {
+          "type": "string",
+          "default": null,
+          "description": "Set the C# fuction module template version, which will be refernced during scaffolding C# function module. null for latest."
+        },
+        "azure-iot-edge.version.javamodule": {
+          "type": "string",
+          "default": null,
+          "description": "Set the Java module template version, which will be refernced during scaffolding Java module. null for latest."
+        },
+        "azure-iot-edge.version.nodemodule": {
```

Comment:
Node.js module template version does not work because users manage them outside VS Code.